packages/java/endpoint/src/main/java/com/vaadin/hilla/route/RouteUtil.java

```diff
@@ -63,8 +60,12 @@ public void setRoutes(
      */
     @Override
     public boolean isRouteAllowed(HttpServletRequest request) {
-        if (registeredRoutes == null) {
-            collectClientRoutes(request);
+        var config = ApplicationConfiguration
+                .get(new VaadinServletContext(request.getServletContext()));
+        boolean isLiveReloadMode = config.getMode()
+                .equals(Mode.DEVELOPMENT_FRONTEND_LIVERELOAD);
```

Comment:
This looks quite heavy and it would be great to determine execution mode once and for all, but let's avoid premature optimization.